nims-notes-things/src/Form.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import TextArea from "./TextArea";
+import Input from "./Input";
+
+const Form = (props) => {
+  const [formData, setFormData] = useState({
+    name: "",
+    date: "",
+    note: "",
+  });
+
+  const [empty, setFormEmpty] = useState({
+    name: false,
+    date: false,
+    note: false,
+  });
+
+  const handleOnChange = (e, key) => {
+    const newForm = {
+      ...formData,
+      [key]: e.target.value,
+    };
+    setFormData(newForm);
+  };
+
+  const handleOnBlur = (e, key) => {
+    empty.map((data) => {
```

Comment:
You wouldn't really map over the error here. You would need to check if the key value is empty and then set that in the state... then depending on the state display an error message. 
See my comment below on line 49

Reply:
Also make sure if you are using a variable in the html (jsx) code you put curly brackets around `<h2>{key.data}</h2>` otherwise it will just display `key.data` on the page and not the value `key.data` is representing

Reply:
You can do something like: 
You are passing e.target.value -> which would be the value coming as an argument and not the event.
```
const handleOnBlur = (value, key) => {
  setFormEmpty((previousState) => {
    return {
      ...previousState,
      [key]: value === "",
    };
  });
};
```

It will take the previous state whatever that would be, and then add the key you are wanting to change/check and will set it to whatever `value === ""` is